add_node.rb

```diff
@@ -1,17 +1,67 @@
 #!/usr/bin/env ruby
 
 require 'json'
+require 'date'
+
+def semester_exists semester
+	possible_semesters = ["mid spring","end spring","mid autumn","end autumn"]
+	if possible_semesters.include? semester
```

Comment:
If `semester` exists, why would you want to return `false`? I'm not sure if I understand. May be you wanted something like this:

```
def semester_exists? semester
  possible_semesters = [ .. ]
  possible_semesters.include? semester
end
```